mount/mount_windows.go

```diff
@@ -47,14 +51,67 @@ func (m *Mount) Mount(target string) error {
 		return errors.Wrapf(err, "failed to activate layer %s", m.Source)
 	}
 	defer func() {
-		if err != nil {
+		if retErr != nil {
 			hcsshim.DeactivateLayer(di, layerID)
 		}
 	}()
 
 	if err = hcsshim.PrepareLayer(di, layerID, parentLayerPaths); err != nil {
 		return errors.Wrapf(err, "failed to prepare layer %s", m.Source)
 	}
+	layerPath, err := hcsshim.GetLayerMountPath(di, layerID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get mount path for layer %s", m.Source)
+	}
+
+	if !strings.HasPrefix(layerPath, "\\\\?\\") {
```

Comment:
I will look back at this to see if there is a better way to do it, but this is checking if the sandbox was mounted as a volume or if the function returned the path to the expanded layer on disk. `\\?\` is the prefix of a volume on Windows, the volume path is of the form `\\?\Volume{bc6dcf0c-7fde-4d12-9853-0d1dafad0b5b}\`. If the expanded layer is returned instead of the volume GUID path, it needs to be mounted via symlink. Will add a comment to explain if I can't find a way that's obvious what it is doing.